### ISSUE TITLE
Romeo crashes on public key set if Stacks node returns 404 for tx status

### ISSUE THREAD

**Issue by stjepangolemac** [stacks-network/sbtc#291]
### Expected Behavior

First of all, Stacks nodes supposedly should never return 404 when asked for a transaction they've just broadcasted. Now, with this behavior the best we can do is consider the transaction pending if it's 404. In our testing the node becomes aware of it, but it might take some time — ~20 minutes in some cases.

### Actual Behavior

Node returns 404 when asked about a transaction, Romeo crashes.

### Steps to Reproduce

Hard to reproduce manually, but at the moment starting Romeo without logs on testnet reliably produces this crash.

**Comment by AshtonStephens** [comment]
The foundation is spinning up a node so we might be able to rely on that one more than the hiro default testnet

**Comment by stjepangolemac** [comment]
Yep that's good, but also good to have this in case we come across the same problem later.

**Comment by AshtonStephens** [comment]
We might be able to do something similar here to what I did with the state: https://github.com/stacks-network/sbtc/pull/290
